lib/elixir/lib/macro.ex

```diff
@@ -232,6 +232,10 @@ defmodule Macro do
     unpipe(right, unpipe(left, acc))
   end
 
+  defp unpipe({:&, _, [{:/, _, [func, 1]}]}, acc) do
+    [{func, 0} | acc]
+  end
```

Comment:
Thank you for the PR. We should handle this in the `pipe` function rather than in `unpipe`. Unpipe is only about breaking it apart. :) We should also improve the error message for `{:&, _, [{:/, _, [func, n]}]} when n != 1` to say you must pipe to a function of arity 1.

Reply:
My point in having it in `unpipe` is to output `{func, n}` in case of `&foo(a, &1, c)` instead of calling lambda. However in this case I can move this to `pipe` as the mentioned case will be only in one situation which can be handled in `unpipe` and all other cases can be handled in `pipe`.

Reply:
It is really unexpected for unpipe to break anything besides pipes apart though. So we need to change it.

Reply:
The other cases shouldn’t be handled in unpipe as well. Unpipe is about breaking pipes into a list without changing the underlying AST.

Reply:
We also shouldn’t expect `&` to be in the format of a function calls it could be any expression, even &[&1], so we need to handle it in a case, as described in the original issue.

Reply:
Yes, but my point is that `&foo(a, &1, b)` translates much better to `{func, 1}` than `{func, 0}` where it is then processed further in the `pipe`. As otherwise the 3rd argument to `pipe` is almost pointless as we do not use it anywhere.